tutorials/euclid_access/Euclid_ERO.md

```diff
@@ -0,0 +1,687 @@
+
+```{code-cell} ipython3
+wcs = WCS(hdr) # VIS
+wcs2 = WCS(hdr2) # NISP
+xy = wcs.all_world2pix(gaia_objects["ra"],gaia_objects["dec"],0)
+xy2 = wcs2.all_world2pix(gaia_objects["ra"],gaia_objects["dec"],0)
+
+gaia_objects["x_vis"] = xy[0]
+gaia_objects["y_vis"] = xy[1]
+gaia_objects["x_nisp"] = xy2[0]
+gaia_objects["y_nisp"] = xy2[1]
+```
+
+We save the Gaia table to disk as we will later use it for the visualization in `Firefly`.
+
+```{code-cell} ipython3
+gaia_objects.write("./data/gaiatable.csv", format="csv", overwrite=True)
+```
+
+Now we can overlay the Gaia sources on the VIS and NISP images (here the x/y coordinates become handy).
+
+```{code-cell} ipython3
+fig = plt.figure(figsize=(20,10))
+ax1 = fig.add_subplot(1,2,1)
+ax2 = fig.add_subplot(1,2,2)
+ax1.imshow(np.log10(img), cmap="Greys", origin="lower")
+ax1.plot(gaia_objects["x_vis"], gaia_objects["y_vis"] , "o", markersize=8 , markeredgecolor="red", fillstyle="none")
+ax1.set_title("VIS")
+
+ax2.imshow(np.log10(img2), cmap="Greys", origin="lower")
+ax2.plot(gaia_objects["x_nisp"], gaia_objects["y_nisp"] , "o", markersize=8 , markeredgecolor="red", fillstyle="none")
+ax2.set_title("NISP")
+
+plt.show()
+```
+
+## Match the Gaia Catalog to the VIS and NISP Catalogs
+
+Now, we match the Gaia source positions to the extracted sources in the VIS and NISP images.
+
+We first define which Gaia columns to copy to the matched catalog as well as the matching distance.
+
+```{code-cell} ipython3
+gaia_keys = ["source_id", "phot_g_mean_mag", "phot_bp_mean_mag", "phot_rp_mean_mag","ra","dec","pmra","pmdec"]
+matching_distance = 0.6*u.arcsecond
+```
+
+First match to the VIS image. We use the `astropy` *SkyCoord()* function for matching in sky coordinates.
+
+```{code-cell} ipython3
+c = SkyCoord(ra=phot["ra_fit"]*u.degree, dec=phot["dec_fit"]*u.degree )
+catalog = SkyCoord(ra=gaia_objects["ra"].data*u.degree, dec=gaia_objects["dec"].data*u.degree)
+idx, d2d, d3d = c.match_to_catalog_sky(catalog)
+
+sel_matched = np.where(d2d.to(u.arcsecond) < (matching_distance))[0]
+print("Gaia Sources matched to VIS: {}".format( len(sel_matched) ) )
+phot["gaia_distance"] = d2d.to(u.arcsecond)
+
+for gaia_key in gaia_keys:
+    phot["gaia_{}".format(gaia_key)] = 0.0
+    phot["gaia_{}".format(gaia_key)][sel_matched] = gaia_objects[gaia_key][idx[sel_matched]]
+```
+
+And then we add the NISP sources. Note that we do not have to perform matching here because by design the VIS and NISP sources are matched (spatial prior forced photometry).
+
+```{code-cell} ipython3
+phot2["gaia_distance"] = d2d.to(u.arcsecond)
+
+for gaia_key in gaia_keys:
+    phot2["gaia_{}".format(gaia_key)] = 0.0
+    phot2["gaia_{}".format(gaia_key)][sel_matched] = gaia_objects[gaia_key][idx[sel_matched]]
+```
+
+Once matched, we can now compare the Gaia and Euclid/NISP magnitudes of the stars.
+
+```{code-cell} ipython3
+# Data
+x = phot["gaia_phot_rp_mean_mag"]
+y = -2.5*np.log10(phot["flux_fit"]) + hdr["ZP_STACK"]
```

Comment:
Narrative says Euclid/NISP magnitudes but code uses `phot["flux_fit"]` which contains Euclid/VIS magnitudes? Doesn't `phot2["flux_fit"]` contains Euclid/NISP magnitudes? If that's correct, then we need to change one or the other.